test/e2e_node/kubelet_authz_test.go

```diff
@@ -0,0 +1,120 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/kubernetes/pkg/cluster/ports"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eauth "k8s.io/kubernetes/test/e2e/framework/auth"
+)
+
+var _ = SIGDescribe("Kubelet Authz", func() {
+	f := framework.NewDefaultFramework("kubelet-authz-test")
+	ginkgo.Context("when calling kubelet API", func() {
+
+		ginkgo.It("check /healthz enpoint is accessible via nodes/healthz RBAC", func(ctx context.Context) {
+			runKubeletAuthzTest(ctx, f, "healthz", "healthz")
+		})
+		ginkgo.It("check /healthz enpoint is accessible via nodes/proxy RBAC", func(ctx context.Context) {
+			runKubeletAuthzTest(ctx, f, "healthz", "proxy")
+		})
+	})
+})
+
+func runKubeletAuthzTest(ctx context.Context, f *framework.Framework, endpoint, authzSubresource string) {
+	ns := "default"
+	saName := authzSubresource
+	crName := authzSubresource
+	verb := "get"
+	resource := "nodes"
+	_, err := f.ClientSet.CoreV1().ServiceAccounts(ns).Create(ctx, &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      saName,
+			Namespace: ns,
+		},
+	}, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	_, err = f.ClientSet.RbacV1().ClusterRoles().Create(ctx, &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crName,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:     []string{verb},
+				Resources: []string{resource + "/" + authzSubresource},
+			},
+		},
+	}, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	subject := rbacv1.Subject{
+		Kind:      "ServiceAccount",
+		Namespace: ns,
+		Name:      saName,
+	}
+
+	err = e2eauth.BindClusterRole(ctx, f.ClientSet.RbacV1(), crName, ns, subject)
+	framework.ExpectNoError(err)
+
+	err = e2eauth.WaitForSubresourceAuthorizationUpdate(
+		ctx,
+		f.ClientSet.AuthorizationV1(),
+		serviceaccount.MakeUsername(ns, saName),
+		ns,
+		verb,
+		authzSubresource,
+		schema.GroupResource{Group: "", Resource: resource},
+		true,
+	)
+	framework.ExpectNoError(err)
+
+	tr, err := f.ClientSet.CoreV1().ServiceAccounts(ns).CreateToken(ctx, saName, &authenticationv1.TokenRequest{}, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	resp, err := healthCheck(fmt.Sprintf("http://127.0.0.1:%d/%s", ports.KubeletHealthzPort, endpoint), tr.Status.Token)
```

Comment:
The healthz port doesn't have authorization.
```suggestion
	resp, err := healthCheck(fmt.Sprintf("https://127.0.0.1:%d/%s", ports.KubeletPort, endpoint), tr.Status.Token)
```

Reply:
Good catch! Done.